docs/source/usermanual.rst

```diff
@@ -0,0 +1,2962 @@
+It sits between the application and a `worker` SPARSE library, marshalling inputs into the backend library and marshalling results back to the application.
+hipSPARSE exports an interface that does not require the client to change, regardless of the chosen backend.
+Currently, hipSPARSE supports rocSPARSE and cuSPARSE as backends.
+hipSPARSE focuses on convenience and portability.
+If performance outweighs these factors, then using rocSPARSE itself is highly recommended.
+rocSPARSE can also be found on `GitHub <https://github.com/ROCmSoftwarePlatform/rocSPARSE/>`_.
+
+.. _hipsparse_building:
+
+Building and Installing
+=======================
+
+Prerequisites
+-------------
+hipSPARSE requires a ROCm enabled platform, more information `here <https://rocm.github.io/>`_.
+
+Installing pre-built packages
+-----------------------------
+hipSPARSE can be installed from `AMD ROCm repository <https://rocm.github.io/ROCmInstall.html#installing-from-amd-rocm-repositories>`_.
+For detailed instructions on how to set up ROCm on different platforms, see the `AMD ROCm Platform Installation Guide for Linux <https://rocm.github.io/ROCmInstall.html>`_.
+
+hipSPARSE can be installed on e.g. Ubuntu using
+
+::
+
+    $ sudo apt-get update
+    $ sudo apt-get install hipsparse
+
+Once installed, hipSPARSE can be used just like any other library with a C API.
+The header file will need to be included in the user code in order to make calls into hipSPARSE, and the hipSPARSE shared library will become link-time and run-time dependent for the user application.
+
+Building hipSPARSE from source
+------------------------------
+Building from source is not necessary, as hipSPARSE can be used after installing the pre-built packages as described above.
+If desired, the following instructions can be used to build hipSPARSE from source.
+Furthermore, the following compile-time dependencies must be met
+
+- `rocSPARSE <https://github.com/ROCmSoftwarePlatform/rocSPARSE>`_
+- `git <https://git-scm.com/>`_
+- `CMake <https://cmake.org/>`_ 3.5 or later
+- `AMD ROCm <https://github.com/RadeonOpenCompute/ROCm>`_
+- `googletest <https://github.com/google/googletest>`_ (optional, for clients)
+
+Download hipSPARSE
+``````````````````
+The hipSPARSE source code is available at the `hipSPARSE GitHub page <https://github.com/ROCmSoftwarePlatform/hipSPARSE>`_.
+Download the master branch using:
+
+::
+
+  $ git clone -b master https://github.com/ROCmSoftwarePlatform/hipSPARSE.git
+  $ cd hipSPARSE
+
+Below are steps to build different packages of the library, including dependencies and clients.
+It is recommended to install hipSPARSE using the `install.sh` script.
+
+Using `install.sh` to build hipSPARSE with dependencies
+```````````````````````````````````````````````````````
+The following table lists common uses of `install.sh` to build dependencies + library.
+
+.. tabularcolumns::
+      |\X{1}{6}|\X{5}{6}|
+
+================= ====
```

Comment:
Does the spacing of === matter here? I.e. should it match the length of Command and Description?